src/components/SomeComponent.tsx

```diff
@@ -1,7 +1,14 @@
+import { useEffect, useState } from 'react'
+
 type SomeComponentProps = {
   someFlag: boolean
 }
 
 export const SomeComponent = ({ someFlag }: SomeComponentProps) => {
-  return <p>ここに、このコンポーネントがレンダリングされた回数を表示してみよう!</p>
+  const [count, setCount] = useState(0)
+  useEffect(() => {
+    someFlag && setCount((prev) => prev + 1)
+  }, [someFlag])
```

Comment:
@k-kbot 
useEffectの第2引数に`[someFlag]` を指定しないと、useEffect が無限ループしてしまいます。

### useEffectの第2引数が[someFlag]の場合
1. Mount時に SomeComponent の useEffect が実行される。ただし親コンポーネントの flag が false のため、useEffect 内のコールバック関数で setState は実行されない
2. 親コンポーネントの button がクリックされたときに 親コンポーネントの flag state が変更される
3. 親コンポーネントが再レンダリングされる
4. 親コンポーネントが再レンダリングされるため、SomeComponent も再レンダリングされる
5. SomeComponent の useEffect が再実行される。flag が true のため、useEffect 内のコールバック関数で setState は実行される
6. SomeComponent の useState が変更されたため、SomeComponent が再レンダリングされる
7. 以降、SomeComponent は無限ループする

また、第2引数に`[someFlag]` を指定しないと、eslint-plugin-react-hooks の react-hooks/exhaustive-deps ルールを導入していると、警告が発生しています。
```
React Hook useEffect contains a call to 'setCount'. Without a list of dependencies, this can lead to an infinite chain of updates. To fix this, pass [someFlag] as a second argument to the useEffect Hook.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)
```
![image](https://user-images.githubusercontent.com/32897182/185778905-8c01504c-333f-48bb-86e8-191d0d468dcf.png)